macros/utils/cross_db_utils/safe_cast.sql

```diff
@@ -1,5 +1,7 @@
 {% macro safe_cast(field, type) %}
-    {% if dbt_version >= '1.2.0' %}
+    {% if target.type == 'databricks' %}
```

Comment:
Mmmm WDYT about creating an adapter for safe cast that by default uses this safe_cast (versioning support) and for databricks returns `try_cast({{field}} as {{type}})`?